xmrstak/misc/executor.cpp

```diff
@@ -509,49 +511,43 @@ void executor::ex_main()
 			win_exit();
 		}
 #endif
-		if(!cfg.tls) tls = false;
-		pools.emplace_back(i+1, cfg.sPoolAddr, cfg.sWalletAddr, cfg.sPasswd, cfg.weight, false, cfg.tls, cfg.tls_fingerprint, cfg.nicehash);
+		if(!cfg.tls) dev_tls = false;
+
+		if(!xmrstak::params::inst().poolURL.empty() && xmrstak::params::inst().poolURL == cfg.sPoolAddr)
```

Comment:
Does this mean that the pass word and username passed by the user via cli params is only used if the user also provide a pool address?

Reply:
Yes. What's the point of sending off a username and password to a random pool?

Reply:
The point is that a miner operator can add a pool to the config and than overwrite only the password which is in many pools the unique identifier for the miner rig.
The PR implementation has a side effect that the user can not see that the password provided via cli is not used in the case where no pool url is provided via a second cli option.

Reply:
If this should be the behavior than we need to add a section somewhere to the documentation and best to the help message.

Reply:
My instinct is to veto that on grounds of security issues where login/pass is actually a login pass. I will add an error instead.